test/external/guardian/liquidationWithdrawalKeyHijacking.ts

```diff
@@ -0,0 +1,479 @@
+      });
+      return withdrawalKey;
+    }
+
+    async function executeProxyLiquidation(
+        _solidAccount: AccountStruct,
+        _liquidAccount: AccountStruct,
+        _withdrawals: any,
+        _deposits: any,
+        _amountWeiForLiquidation: any,
+        _wethAmount: any
+      ) {
+      // Give the contract the WETH needed to complete the exchange
+      const testTrader = await impersonate(core.testEcosystem!.testExchangeWrapper.address, true, _wethAmount.mul(10));
+      await setupWETHBalance(
+        core,
+        testTrader,
+        _wethAmount.mul(5),
+        { address: '0x000000000000000000000000000000000000dead' },
+      );
+
+      const allKeys = _withdrawals.concat(_deposits);
+      const tradeTypes = Array(
+        _withdrawals.length
+        ).fill(UnwrapperTradeType.FromWithdrawal).concat(
+          Array(
+            _deposits.length
+            ).fill(UnwrapperTradeType.FromWithdrawal))
+
+      const liquidationData = ethers.utils.defaultAbiCoder.encode(
+        ['uint8[]', 'bytes32[]'],
+        [tradeTypes, allKeys],
+      );
+      const withdrawals = await Promise.all(withdrawalKeys.map(key => unwrapper.getWithdrawalInfo(key)));
+      const deposit = depositKey ? await wrapper.getDepositInfo(depositKey) : undefined;
+      const allStructs = withdrawals
+        .map(w => ({ inputAmount: w.inputAmount, outputAmount: w.outputAmount }))
+        .concat(deposit ? [{ inputAmount: deposit.outputAmount, outputAmount: deposit.inputAmount }] : []);
+      const outputAmountForSwap = allStructs
+        .reduce((acc, struct) => {
+          if (acc.input.gt(ZERO_BI)) {
+            const inputAmount = acc.input.lt(struct.inputAmount)
+              ? acc.input
+              : struct.inputAmount;
+            const outputAmount = acc.input.lt(struct.inputAmount)
+              ? struct.outputAmount.mul(acc.input).div(struct.inputAmount)
+              : struct.outputAmount;
+
+            acc.input = acc.input.sub(inputAmount);
+            acc.output = acc.output.add(outputAmount);
+          }
+          return acc;
+        }, { output: ZERO_BI, input: _amountWeiForLiquidation })
+        .output;
+
+      const zapParam = await getLiquidateIsolationModeZapPath(
+        [marketId, core.marketIds.nativeUsdc!, core.marketIds.weth],
+        [_amountWeiForLiquidation, outputAmountForSwap, _wethAmount],
+        unwrapper,
+        core,
+      );
+      zapParam.tradersPath[0].tradeData = liquidationData;
+      await liquidateV4WithZapParam(
+        core,
+        _solidAccount,
+        _liquidAccount,
+        zapParam,
+      );
+    }
+
+    async function getMarketBalances(account: any) {
+      const balances = new Map<string, BigNumber>();;
+      const markets = await core.dolomiteMargin.getAccountMarketsWithBalances(account);
+      for (const market of markets) {
+        const par = await core.dolomiteMargin.getAccountPar(account, market.toNumber());
```

Comment:
Why not use `await core.dolomiteMargin.getAccountWei(account, marketId)` as in `expectProtocolBalance`?